USE_CASES.md

```diff
@@ -1000,3 +1001,90 @@ namespace TwilioTest
     }
 }
 ```
+
+<a name="working-with-permissions" ></a>
+# Working with permissions
+
+The permissions builder is a convenient way to manipulate API key permissions when creating new API keys or managing existing API keys. 
+
+You can enums named according to the various [permissions](https://sendgrid.api-docs.io/v3.0/api-key-permissions) to add the scopes required for those permissions. 
+
+By default, all scopes for a given permission are added; however, You can filter out certain scopes by passing a ScopeOptions parameter or adding some filtering functions.
+
+For example, to create an API key for all *Alerts* scopes and read only *Marketing Campaigns*:
+
+```
+var apiKey = Environment.GetEnvironmentVariable("NAME_OF_THE_ENVIRONMENT_VARIABLE_FOR_YOUR_SENDGRID_KEY");
+var client = new SendGridClient(apiKey);
+var builder = new SendGridPermissionsBuilder();
+builder.AddPermissionsFor(SendGridPermission.Alerts);
+builder.AddPermissionsFor(SendGridPermission.MarketingCampaigns, ScopeOptions.ReadOnly);
+
+/*
+The above builder will emit the following scopes:
+
+alerts.create
+alerts.delete
+alerts.read
+alerts.update
+marketing_campaigns.read
+*/
+
+// now use the provided extension method to create an API key
+
+await client.CreateApiKey(builder, "Alerts & Read-Only Marketing Campaigns API Key");
+```
+
+There are also some methods to easily create API keys for various common use cases.
+
+For example, to create a Mail API:
```

Comment:
"create a Mail API" to something like "create a Mail Send API Key scoped for sending email"